chain-evm/src/machine.rs

```diff
@@ -568,19 +592,27 @@ impl<'a, State: EvmState> StackState<'a> for VirtualMachine<'a, State> {
     }
 
     fn set_code(&mut self, address: H160, code: Vec<u8>) {
-        self.substate.account_mut(address, self.state).code = code;
+        self.substate.account_mut(address, self.state).state.code = code;
     }
 
     fn transfer(&mut self, transfer: Transfer) -> Result<(), ExitError> {
         let source = self.substate.account_mut(transfer.source, self.state);
 
-        source.balance = match source.balance.checked_sub(transfer.value) {
+        source.balance = match source
+            .balance
+            .checked_sub(transfer.value.try_into().map_err(|_| ExitError::OutOfGas)?)
```

Comment:
Can the failed `try_into()` be caused by value being bigger than what a balance can hold? In that case I think it's better to have a more descriptive error message, since I'm sure this can happen in a lot of other places as well

Reply:
Yes, it is possible, we can throw something similar with `ExitError::Other("Balance overflow".into()))`, but the purpose of throwing `ExitError::OutOfGas` it is if value which we want to spend is overflowed it is obviously also means that you are trying to spend more than you have ``ExitError::OutOfGas` error in this case. 
So IMO it is not a huge difference between those two erros types, but I agree `ExitError::Other("Balance overflow".into()))` reflects the core problem, so lets throw such error.

Reply:
>  it is if value which we want to spend is overflowed it is obviously also means that you are trying to spend more than you have

Then wouldn't `OutOfFund` be more appropriate? `OutOfGas` has a different meaning